spec/shard.md

```diff
@@ -0,0 +1,605 @@
+│(8 bytes)│offset   │offset   │offset   │num_entry│offset   │num_entry│offset   │num_entry│
+│         │(8 bytes)│(8 bytes)│(8 bytes)│(8 bytes)│(8 bytes)│(8 bytes)│(8 bytes)│(8 bytes)│
+└─────────┴─────────┴─────────┴─────────┴─────────┴─────────┴─────────┴─────────┴─────────┘
+0         8        16        24        32        40        48        56        64        72
+
+┌─────────────────────────────────────┬─────────┬─────────┬─────────────────────────────────────┐
+│        chunk_hash_hmac_key          │creation │key_expiry│           _buffer (reserved)        │
+│             (32 bytes)              │timestamp│(8 bytes)│             (48 bytes)              │
+│                                     │(8 bytes)│         │                                     │
+└─────────────────────────────────────┴─────────┴─────────┴─────────────────────────────────────┘
+72                                  104       112       120                                   168
+
+┌─────────┬─────────┬─────────┬─────────┐
+│stored_  │material │stored_  │footer_  │
+│bytes_on │ized_    │bytes    │offset   │
+│disk     │bytes    │(8 bytes)│(8 bytes)│
+│(8 bytes)│(8 bytes)│         │         │
+└─────────┴─────────┴─────────┴─────────┘
+168      176       184       192
+```
+
+**Deserialization steps**:
+
+1. Seek to `file_size - footer_size`
+2. Read all fields sequentially as u64 values
+3. Verify version equals 1
+
+### 3. File Info Section
+
+**Location**: `footer.file_info_offset` to `footer.cas_info_offset`
+
+This section contains a sequence of 0 or more file information blocks, each consisting of:
+
+#### FileDataSequenceHeader
+
+```rust
+struct FileDataSequenceHeader {
+    file_hash: [u64; 4],  // 32-byte file hash
+    file_flags: u32,      // Flags indicating conditional sections that follow
+    num_entries: u32,     // Number of FileDataSequenceEntry structures
+    _unused: [u8; 8],     // Reserved space 8 bytes
+}
+```
+
+**Memory Layout**:
+
+```txt
+┌────────────────────────────────────────────────────────────────┬──────────┬──────────┬──────────┐
+│                       file_hash (32 bytes)                    │file_flags│num_entries│ _unused  │
+│                        File Hash Value                        │(4 bytes) │(4 bytes) │(8 bytes) │
+└────────────────────────────────────────────────────────────────┴──────────┴──────────┴──────────┘
+0                                                              32         36         40         48
+```
+
+**File Flags**:
+
+- `MDB_FILE_FLAG_WITH_VERIFICATION` (0x80000000 or 1 << 31): Has verification entries
+- `MDB_FILE_FLAG_WITH_METADATA_EXT` (0x40000000 or 1 << 30): Has metadata extension
+
+Given the `file_data_sequence_header.file_flags & MASK` (bitwise AND) operations, if the result != 0 then the effect is true.
+
+#### FileDataSequenceEntry
+
+```rust
+struct FileDataSequenceEntry {
+    cas_hash: [u64; 4],           // 32-byte CAS block hash
+    cas_flags: u32,               // CAS flags
```

Comment:
Again, we should probably list out what valid flags are and where they're located in this bitmap (like in the `FileDataSequenceHeader` above.